sklearn/cluster/hierarchical.py

```diff
@@ -712,6 +712,10 @@ def fit(self, X, y=None):
         if isinstance(memory, six.string_types):
             memory = Memory(cachedir=memory, verbose=0)
 
+        if self.n_clusters <= 0:
+            raise ValueError("%s was provided as n_clusters. It should"
```

Comment:
I would first say what is wrong, then what was provided, but that might be a matter of taste.